### DISCUSSION TITLE
Systray with Fly manager

### DISCUSSION THREAD

**Discussion by Dmitriy147** [fyne-io/fyne#4383] [General]
My app use systray with menu on v.2.4.1. After 	starting my program i can see  gui form and then , instead of system tray icon appeared, i can see error "systray error: failed to register: the name org.kde.StatusNotifierWatcher was not provided by any .service files".
I have os AstraLinux 1.7, fly, no kde.
Is it possible to have tray icon in my os configuration?

**Comment by Jacalz** [comment]
As far as I know, no one on our team uses the fly window manager so you would have to google on how to how to (if possible) to install a system tray for it. Worst case, I guess you need a different desktop environment.

**Comment by Dmitriy147** [comment]
I have managed to build app with system tray in Fly wm by using "https://github.com/getlantern/systray" package...
But its need to install:
sudo apt-get install gcc libgtk-3-dev libayatana-appindicator3-dev

**Reply by andydotxyz** [reply]
That systray package uses the legacy ("xembed") system tray - we are using the new DBUS based system. Your window manager may need a plugin if it is not already integrated.

**Reply by Dmitriy147** [reply]
Unfortunately dbus.1.12.24-0+deb11u1 is installed and systray error is presents - the name org..kde.StatusNotifierWatcher was not provided by any .service files.

**Reply by andydotxyz** [reply]
I was not referring to a dbus dependency, but a window manager module that actually hosts system tray.
That message means your window manager does not have a system tray, so you'll need to hunt down a module to enable it.
Or use a standalone tray like https://github.com/andydotxyz/fray
